src/biguint.rs

```diff
@@ -434,6 +434,47 @@ impl One for BigUint {
 
 impl Unsigned for BigUint {}
 
+macro_rules! pow_impl {
+    ($unsigned:ty) => {
+         impl<'a> pow::Pow<$unsigned> for &'a BigUint {
+            type Output = BigUint;
+
+            #[inline]
+            fn pow(self, mut exp: $unsigned) -> Self::Output {
+                // square and multiply from num_traits::pow in the next branch
+                let mut base = self.clone();
+
+                if exp == 0 { return BigUint::one(); }
```

Comment:
Might as well check this first, to potentially avoid a useless clone.

Reply:
Good point.